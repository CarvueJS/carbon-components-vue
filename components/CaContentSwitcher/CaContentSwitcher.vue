<template>
  <div data-content-switcher class="bx--content-switcher">
    <slot></slot>
  </div>
</template>

<script>
  import { ContentSwitcher } from "carbon-components"

  export default {
    name: 'ca-content-switcher',
    mounted() {
      this.$el.addEventListener('content-switcher-selected', (e)=>{
        this.$emit('selected', e.detail.item.getAttribute('switchvalue'))
      });
    }
  };
</script>

<style></style>
