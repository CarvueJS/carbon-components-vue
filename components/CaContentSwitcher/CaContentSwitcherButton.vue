<template>
  <button class="bx--content-switcher-btn" data-target=".demo--panel--opt-2" :class="dynamicClass" :switchValue="switchValue">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'ca-content-switcher-button',
  props: {
    selected: {
      type: Boolean,
      default: false,
    },
    switchValue: {
      type: String,
      default: '',
    },
  },
  computed: {
    dynamicClass() {
      return {
        'bx--content-switcher--selected': this.selected,
      };
    },
  },
};
</script>

<style></style>
