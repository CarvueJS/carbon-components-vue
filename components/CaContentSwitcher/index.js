import CaContentSwitcher from './CaContentSwitcher.vue';
import CaContentSwitcherButton from './CaContentSwitcherButton.vue';

CaContentSwitcher.install = (Vue) => {
  Vue.component(CaContentSwitcher.name, CaContentSwitcher);
  Vue.component(CaContentSwitcherButton.name, CaContentSwitcherButton);
};

export default CaContentSwitcher;
