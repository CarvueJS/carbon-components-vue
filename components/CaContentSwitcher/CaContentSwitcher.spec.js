import { shallow } from 'vue-test-utils';
import CaContentSwitcher from './CaContentSwitcher.vue';
import CaContentSwitcherButton from './CaContentSwitcherButton.vue';

describe('CaContentSwitcher', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaContentSwitcher, { 
      slots: {
        default: CaContentSwitcherButton
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
