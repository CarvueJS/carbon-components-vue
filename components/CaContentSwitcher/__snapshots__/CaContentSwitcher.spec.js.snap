// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaContentSwitcher Mathes snapshot 1`] = `
<div data-content-switcher="" class="bx--content-switcher">
  <button data-target=".demo--panel--opt-2" switchvalue="" class="bx--content-switcher-btn"></button>
</div>
`;
